src/lib/docs/outline-items-from-headings.ts

```diff
@@ -0,0 +1,33 @@
+import { OutlineLinkItem } from 'components/outline-nav/types'
+import { filterTableOfContentsHeadings } from 'components/table-of-contents/utils/filter-table-of-contents-headings'
+
+/**
+ * Heading type coming from our remark anchor-links plugin.
+ */
+export interface AnchorLinksPluginHeading {
+	title: string
+	slug: string
+	level: 1 | 2 | 3 | 4 | 5 | 6
+}
+
+/**
+ * Given a flat array of headings,
+ * Return an array of outline items, for use with OutlineNav components.
+ */
+function outlineItemsFromHeadings(
```

Comment:
🙋🏻 I might be missing some context, won't views like `TutorialView` need this functionality as well? Not just docs?

Reply:
Ah true! I've moved this util into `components/outline-nav/utils` - hopefully that works! Happy to move it again in a subsequent PR if needed.